packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -312,6 +412,137 @@ class CupertinoSliverNavigationBar extends StatelessWidget {
   }
 }
 
+class _CupertinoLargeTitleNavigationBarSliverDelegate
+    extends SliverPersistentHeaderDelegate with DiagnosticableTreeMixin {
+  _CupertinoLargeTitleNavigationBarSliverDelegate({
+    @required this.persistentHeight,
+    @required this.largeTitle,
+    this.leading,
+    this.automaticallyImplyLeading,
+    this.previousPageTitle,
+    this.middle,
+    this.trailing,
+    this.padding,
+    this.border,
+    this.backgroundColor,
+    this.actionsForegroundColor,
+  }) : assert(persistentHeight != null);
+
+  final double persistentHeight;
+
+  final Widget largeTitle;
+
+  final Widget leading;
+
+  final bool automaticallyImplyLeading;
+
+  final String previousPageTitle;
+
+  final Widget middle;
+
+  final Widget trailing;
+
+  final EdgeInsetsDirectional padding;
+
+  final Color backgroundColor;
+
+  final Border border;
+
+  final Color actionsForegroundColor;
+
+  @override
+  double get minExtent => persistentHeight;
+
+  @override
+  double get maxExtent => persistentHeight + _kNavBarLargeTitleHeightExtension;
+
+  @override
+  Widget build(BuildContext context, double shrinkOffset, bool overlapsContent) {
+    final bool showLargeTitle = shrinkOffset < maxExtent - minExtent - _kNavBarShowLargeTitleThreshold;
+
+    final _CupertinoPersistentNavigationBar persistentNavigationBar =
+        new _CupertinoPersistentNavigationBar(
+      leading: leading,
+      automaticallyImplyLeading: automaticallyImplyLeading,
+      previousPageTitle: previousPageTitle,
+      middle: middle ?? largeTitle,
+      trailing: trailing,
+      // If middle widget exists, always show it. Otherwise, show title
+      // when collapsed.
+      middleVisible: middle != null ? null : !showLargeTitle,
+      padding: padding,
+      actionsForegroundColor: actionsForegroundColor,
+    );
+
+    return _wrapWithBackground(
+      border: border,
+      backgroundColor: backgroundColor,
+      child: new Stack(
+        fit: StackFit.expand,
+        children: <Widget>[
+          new Positioned(
+            top: persistentHeight,
+            left: 0.0,
+            right: 0.0,
+            bottom: 0.0,
+            child: new ClipRect(
```

Comment:
not really :S

Unfortunately GitHub doesn't show code moves. I just moved this block so this file flows better. 

The large title actually needs to clip. I can't just hide it behind other things because the thing on top of it is translucent.

Reply:
ah well